sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreChangelog.scala

```diff
@@ -351,24 +351,31 @@ class StateStoreChangelogReaderFactory(
     case f: FileNotFoundException =>
       throw QueryExecutionErrors.failedToReadStreamingStateFileError(fileToRead, f)
   }
-  protected val input: DataInputStream = decompressStream(sourceStream)
+  protected var input: DataInputStream = decompressStream(sourceStream)
+  protected var version: Short = -1
 
   private def readVersion(): Short = {
```

Comment:
Thanks for the follow up!

Can we just remove this `protected var version: Short = -1` and change this line to 
`private lazy val readVersion: Short = {`